### PR TITLE
Fix 559

### PR DESCRIPTION
**Pull request by FFroehlich**

### CODE REVIEW

File: src/model.cpp [hunk 1]
```diff
@@ -1571,7 +1571,6 @@ int Model::checkFinite(const int N, const realtype *array, const char *fun) cons
     auto result = amici::checkFinite(N, array, fun);
 
     if(result != AMICI_SUCCESS) {
-        amici::checkFinite(ts.size(), ts.data(), "ts");
```
Comment (dweindl) [inline src/model.cpp]: When using adjoints, a check for inf might be appropriate. Not sure if that is done in any other place

File: include/amici/defines.h [hunk 2]
```diff
@@ -19,6 +19,7 @@ constexpr double pi = 3.14159265358979323846;
 
 /* Return codes */
 #define AMICI_RECOVERABLE_ERROR          1
+#define AMICI_UNRECOVERABLE_ERROR       -1
```
Comment (dweindl) [inline include/amici/defines.h]: Is there a specific reason to have AMICI_UNRECOVERABLE_ERROR and AMICI_TOO_MUCH_WORK indistinguishable?

File: src/misc.cpp [hunk 3]
```diff
@@ -7,21 +7,26 @@
...
+        if (isInf(array[idx])) {
+            warnMsgIdAndTxt(
+                "AMICI:Inf",
+                "AMICI encountered an Inf value at index %i of %i in %s!", idx,
+                N, fun);
+            return AMICI_RECOVERABLE_ERROR;
         }
     }
-    return(AMICI_SUCCESS);
+    return (AMICI_SUCCESS);
```
Comment (dweindl) [inline src/misc.cpp]:
```suggestion
    return AMICI_SUCCESS;
```

File: include/amici/defines.h (hunk 2 above)
Comment (FFroehlich) [reply include/amici/defines.h]: no, value of `AMICI_UNRECOVERABLE_ERROR` does not matter, just needs to be strictly negative.

File: src/model.cpp (hunk 1 above)
Comment (FFroehlich) [reply src/model.cpp]: hmm good point!

**Pull request merged**